spec/output/generator_spec_rust/block_comments.x/MyXDR.rs

```diff
@@ -232,29 +232,53 @@ where
     #[cfg(feature = "std")]
     fn read_xdr(r: &mut impl Read) -> Result<Self>;
 
+    #[cfg(feature = "std")]
+    fn read_xdr_to_end(r: &mut impl Read) -> Result<Self> {
+        let s = Self::read_xdr(r)?;
+        // Check that any further reads, such as this read of one byte, read no
+        // data, indicating EOF. If a byte is read the data is invalid.
+        if r.read(&mut [0u8; 1])? == 0 {
```

Comment:
I'm not sure this matches up with the behavior of `std::io::Read::read_to_end`. Specifically, the [documentation](https://doc.rust-lang.org/std/io/trait.Read.html#method.read_to_end) reads

> This function will continuously call [read()](https://doc.rust-lang.org/std/io/trait.Read.html#tymethod.read) to append more data to buf until [read()](https://doc.rust-lang.org/std/io/trait.Read.html#tymethod.read) returns either [Ok(0)](https://doc.rust-lang.org/std/result/enum.Result.html#variant.Ok) or an error of non-[ErrorKind::Interrupted](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.Interrupted) kind.

but this does not handle `ErrorKind::Interrupted`. I've never worked with this trait before so I'm not really sure of the significance here.

Reply:
Thanks, that's a good observation. I had read about the error and decided to treat it also as an error because I thought that's what the behavior of the function was, but I think I was reading the docs for `read` rather than `read_to_end` in that moment because a re-read yes definitely say it continues reading when interrupted. I'll update our implementation to match. I've reviewed the source for `read_to_end` as well now.

Reply:
I've written tests for interrupted in https://github.com/stellar/rs-stellar-xdr/pull/178, and all read_xdr implementations continue on `Interrupted`, so this is handled by that. As part of this PR I'm adding docs to the read functions describing their behaviors, and I've added a note to all of the functions saying that they are required to internally continue on interrupted.